### PR TITLE
Dev quant squeeze

### PR DESCRIPTION
**Pull request by ScXfjiang**
The PR was submitted to add the Squeeze/Unsqueeze operation to QuantTensor. Issue: #891

**Changes**
* [x] Add Squeeze/Unsqueeze operation to QuantTensor.
* [x] For meta-data sensitive operations in QuantTensor (e.g., reshape, flatten, transpose, permute, squeeze, unsqueeze), add per-channel granularity test cases.

See more discussions in #891 #728.

### CODE REVIEW

**Comment by ScXfjiang** [comment]
have formatted the code and added a quick fix @Giuseppe5

**Comment by ScXfjiang** [comment]
Hi @Giuseppe5,

I've updated the code to be compatible with PyTorch < 2.0, and it should now pass the CI checks.

In PyTorch < 2.0, torch.squeeze() doesn't support multiple dims.

> Changelog of PyTorch 2.0:
> Update torch.squeeze to allow squeezing multiple dimensions at once (https://github.com/pytorch/pytorch/pull/89017)

To make Brevitas flexible, QuantTensor.squeeze() always supports multiple dims, even when using PyTorch < 2.0.

The method seems more complex than other meta-data sensitive methods because it needs to guarantee that the value tensor and the meta-data tensors choose the same dims to squeeze.

File: src/brevitas/quant_tensor/int_quant_tensor.py [hunk 1]
```diff
@@ -212,6 +212,41 @@ def permute(self, *args, **kwargs):
...
+                raise RuntimeError("no dim specified")
+            arg_dims = [dim if dim >= 0 else dim + len(self.value.shape) for dim in arg_dims]
+            sorted_target_dims = sorted(list(set(one_dims) & set(arg_dims)), reverse=True)
+        value = self.value
+        for dim in sorted_target_dims:
+            value = value.squeeze(dim)
+        tensor_meta = {
+            'scale': self.scale, 'zero_point': self.zero_point, 'bit_width': self.bit_width}
+        for k, tm in tensor_meta.items():
+            if tm is not None and len(value.shape) == len(tm.shape) - len(sorted_target_dims):
```
Comment (Giuseppe5) [inline src/brevitas/quant_tensor/int_quant_tensor.py:235]: metadatas cannot be None anymore

File: src/brevitas/quant_tensor/torch_handler.py [hunk 2]
```diff
@@ -46,6 +46,21 @@ def transpose_handler(inp, *args, **kwargs):
     return inp.transpose(*args, **kwargs)
 
 
+@implements(torch.permute)
+def permute_handler(inp, *args, **kwargs):
+    return inp.permute(*args, **kwargs)
```
[the remark this answers is not in the document]
Comment (ScXfjiang) [reply src/brevitas/quant_tensor/torch_handler.py:51]: Sorry for the late reply. I will update my code next week : )

**Pull request closed without merging**